test/integration/hds_integration_test.cc

```diff
@@ -501,20 +490,17 @@ TEST_P(HdsIntegrationTest, TwoEndpointsSameLocality) {
   host2_stream_->encodeHeaders(Http::TestHeaderMapImpl{{":status", "200"}}, false);
   host2_stream_->encodeData(1024, true);
 
-  // Make sure the cluster processed the endpoints' response
-  test_server_->waitForCounterGe("cluster.anna.health_check.failure", 1);
-  test_server_->waitForCounterGe("cluster.anna.health_check.success", 1);
-
-  // Envoy reports back to server
+  // Receive updates until the one we expect arrives
   ASSERT_TRUE(hds_stream_->waitForGrpcMessage(*dispatcher_, response_));
+  while (!(checkEndpointHealthResponse(response_.endpoint_health_response().endpoints_health(0),
+                                       envoy::api::v2::core::HealthStatus::UNHEALTHY,
+                                       host_upstream_->localAddress()) &
```

Comment:
Nit: prefer logical `&&` over bitwise `&` when expressing a logical condition, i.e. more declarative; also safer when when one of the conjuncts might not be only 0 or 1.